lib/tasks/serve.js

```diff
@@ -52,13 +52,16 @@ module.exports = Task.extend({
       expressServer: expressServer
     });
 
-    return Promise.all([
-        liveReloadServer.start(options),
-        expressServer.start(options)
-      ]).then(function() {
+    // note Promise.all will hang under certain cirumstances, aka
```

Comment:
when we launch the express server, we also use the getPort check, just before calling createServer(or whatever) - if this check fails, I throw. Promise.allSettled may fix this(bluebird has this sort of function but don't recall rsvp)? - but Promise.all does not catch the exception I throw when the port is taken

this is different the live reload case where we just bump up to the next port. in the express server we want to shutdown b/c the port they are expecting to open should not as users would not notice that we instead bumped it(in the console output)

related to this is the test port - this does not port scan - this is something i think we should add to this pr